demo/client/backend/src/fabricClient.js

```diff
@@ -0,0 +1,477 @@
+      throw (result);
+    }
+
+    //  console.log ("<<< Exit: function, enrollUser");
+}
+
+
+//  function setUserContext
+//  Purpose:  to set the context to the user (who called this api)
+//            All subsequent calls using that gateway / contract will be on this user's behalf.
+//  Input:  userName - which has been registered;  If not enrolled, this function enrolls.
+//  Output:  if not error returns SUCCESS else throws an exception
+//          Also, (Global variable) contract will be set to this user's context
+//          All further transactions using this contract will be submitted using
+//          this userName
+async function setUserContext (userName, pwd) {
+    console.log ('>>> Entry: function setUserContext ....');
+
+    try {
+        //  verify if user is registered
+        await enrollUser(userName, pwd);
+    } catch (error) {
+      throw (error);
+    }
+
+    try {
+        // Set connection options; identity and wallet
+        let connectionOptions = {
+            identity: userName,  //  * * * set context to this userName   * * *
+            wallet: wallet,
+            discovery: { enabled: false, asLocalhost: bLocalHost }
+        };
+
+        // Connect to gateway using application specified parameters
+        //  using local variable, userGateway
+        console.log('Connect to Fabric gateway for userName: ' + userName);
+        let userGateway = new Gateway ();
+        await userGateway.connect(connectionProfile, connectionOptions);
+
+        console.log ("Connected to gateway");
+        network =  await userGateway.getNetwork (configdata["channel_name"]);
+        console.log ("Connected to channel, " + configdata["channel_name"]);
+
+        //  contract is global variable, used by invoke and query functions
+        contract =  await network.getContract (configdata["smart_contract_name"]);
+        console.log ("Connected to chaincode, ", configdata["smart_contract_name"]);
+        console.log ('...........................');
+        //  return success
+        var result =
+        {
+          "status": {  'rc': SUCCESS,
+          'msg': 'Successfully set user context to  user:  ' + userName }
+        };
+        return result;
+        }
+    catch (error) {
+        var result =
+        {
+          "status": {  'rc': FAILURE,
+          'msg': 'setUserContext failed for userName,' + userName + ' : ' + error }
+        };
+        console.error('setUserContext failed for userName,' + userName );
+        throw (result);
+      }
+}  //  end of UserContext(userName)
+
+async function invoke(userName, txName, ...args) {
+    console.log (">>> In function, fabricClient:invoke... ");
+    console.log ("userName =***" + userName + "***");
+    console.log ("txName =  ", txName);
+    console.log ("args =  ", args);
+
+    try {
+    await setUserContext (userName,secret);
+    }  catch (error) {
+        throw (error);
+    }
```

Comment:
is this the correct way to return the error? I am not an javascript expert but this just does look good/right?!

Reply:
Added a console.log to error handling code.  That was missing here.